certora/specs/SocialRecoveryModule.spec

```diff
@@ -1,26 +1,176 @@
+
+definition SENTINEL() returns address = 1;
 
 methods {
+    // Social Recovery Module Functions
+    function isGuardian(address, address) external returns (bool) envfree;
+    function guardiansCount(address) external returns (uint256) envfree;
+    function threshold(address) external returns (uint256) envfree;
+
+    // Guardian Storage Functions
+    function guardianStorageContract.getGuardiansCount(address) external returns (uint256) envfree;
+
     // Safe Functions
-    function safeContract.isModuleEnabled(address) external returns (bool) envfree;
+    function safeContract.isModuleEnabled(address module) external returns (bool) envfree;
+    function safeContract.isOwner(address owner) external returns (bool) envfree;
+
+    // Wildcard Functions (Because of use of ISafe interface in Social Recovery Module)
+    function _.isModuleEnabled(address module) external => safeIsModuleEnabled(module) expect bool ALL;
+    function _.isOwner(address owner) external => safeIsOwner(calledContract, owner) expect bool ALL;
 }
 
-// A setup function that requires Safe contract to enabled the Social Recovery
-// Module.
+// Ghost variable to store the guardian mapping of a Safe.
+ghost mapping(address => mapping(address => address)) ghostGuardians {
+    init_state axiom
+        forall address safeAccount.
+        forall address guardian.
+            ghostGuardians[safeAccount][guardian] == 0;
+}
+hook Sload address value guardianStorageContract.entries[KEY address safeAccount].guardians[KEY address guardian] {
+    require ghostGuardians[safeAccount][guardian] == value;
+}
+hook Sstore guardianStorageContract.entries[KEY address safeAccount].guardians[KEY address guardian] address value {
+    ghostGuardians[safeAccount][guardian] = value;
+}
+
+// Ghost variable to store the guardian count of a Safe.
+ghost mapping(address => mathint) ghostGuardianCount {
+    init_state axiom forall address safeAccount. ghostGuardianCount[safeAccount] == 0;
+}
+hook Sload uint256 value guardianStorageContract.entries[KEY address safeAccount].count {
+    require ghostGuardianCount[safeAccount] == to_mathint(value);
+}
+hook Sstore guardianStorageContract.entries[KEY address safeAccount].count uint256 value {
+    ghostGuardianCount[safeAccount] = value;
+}
+
+// Ghost variable to store the threshold of a Safe.
+ghost mapping(address => mathint) ghostThreshold {
+    init_state axiom forall address safeAccount. ghostThreshold[safeAccount] == 0;
+}
+hook Sload uint256 value guardianStorageContract.entries[KEY address safeAccount].threshold {
+    require ghostThreshold[safeAccount] == to_mathint(value);
+}
+hook Sstore guardianStorageContract.entries[KEY address safeAccount].threshold uint256 value {
+    ghostThreshold[safeAccount] = value;
+}
+
+// A summary function to check if a module is enabled in the Safe contract.
+function safeIsModuleEnabled(address module) returns bool {
+    return safeContract.isModuleEnabled(module);
```

Comment:
Personally, I find this function not very useful at all, and typing the function out is 7 characters more.

Reply:
This is used as a summary for the wildcard function (due to `ISafe` use in the Module.)

If I try to summarize directly, it gives me the error:
```
Only CVL constructs (e.g. CVL functions or ghosts) can be used as summaries.
```

Reply:
Rectified, both now checks callee 👍🏾